File: app/server/project/src/main/java/vsu/csf/rentyserver/model/entity/Receipt.java
package vsu.csf.rentyserver.model.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import lombok.experimental.Accessors;
import org.springframework.data.annotation.CreatedDate;

import java.time.OffsetDateTime;

@Entity
@Getter
@Setter
@Accessors(chain = true)
@Table(name = "receipt")
public class Receipt {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(name = "receipt_id")
    private String receiptId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id")
    private AppUser user;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "employee_id")
    private AppUser employee;

    @CreatedDate
    @Column(name = "created_at")
    private OffsetDateTime createdAt;

    @Column(name = "pay_link")
    private String payLink;

    @Column(name = "status")
    private String status;

}

File: app/client/Renty-client/lib/api/dto/response/receipt.dart
class ReceiptResponse {
  final int receiptId;
  final int user;
  final int employee;
  final String payLink;
  final String createdAt;
  final double sum;
  final String status;

  ReceiptResponse({
    required this.receiptId,
    required this.user,
    required this.employee,
    required this.payLink,
    required this.createdAt,
    required this.sum,
    required this.status,
  });

  factory ReceiptResponse.fromJson(Map<String, dynamic> json) {
    return ReceiptResponse(
      receiptId: json['receipt_id'],
      user: json['user'],
      employee: json['employee'],
      payLink: json['pay_link'],
      createdAt: json['created_at'],
      sum: json['sum'],
      status: json['status'],
    );
  }
}
